### PR TITLE
ORC-443: [C++] Code improvements in ColumnWriter.

### PR DESCRIPTION
**Pull request by fangzheng**
1. in add() function, re-order code to verify input parameters before modifying any internal state.

2. in add() function, move the calls to colIndexStatistics->increase(1) out of the loop.

### CODE REVIEW

File: c++/src/ColumnWriter.cc [hunk 1]
```diff
@@ -762,25 +770,26 @@ namespace orc {
...
       throw InvalidArgument("Failed to cast to DoubleVectorBatch");
     }
-
-    const double* doubleData = dblBatch->data.data() + offset;
-    const char* notNull = dblBatch->hasNulls ?
-                          dblBatch->notNull.data() + offset : nullptr;
-
     DoubleColumnStatisticsImpl* doubleStats =
-      dynamic_cast<DoubleColumnStatisticsImpl*>(colIndexStatistics.get());
+        dynamic_cast<DoubleColumnStatisticsImpl*>(colIndexStatistics.get());
```
Comment (wgtmac) [inline c++/src/ColumnWriter.cc]: please use same indent

File: c++/src/ColumnWriter.cc [hunk 2]
```diff
@@ -1477,22 +1508,25 @@ namespace orc {
...
     if (decBatch == nullptr) {
       throw InvalidArgument("Failed to cast to Decimal64VectorBatch");
     }
 
-    const char* notNull = decBatch->hasNulls ?
-                          decBatch->notNull.data() + offset : nullptr;
-    const int64_t* values = decBatch->values.data() + offset;
     DecimalColumnStatisticsImpl* decStats =
-      dynamic_cast<DecimalColumnStatisticsImpl*>(colIndexStatistics.get());
+        dynamic_cast<DecimalColumnStatisticsImpl*>(colIndexStatistics.get());
```
Comment (wgtmac) [inline c++/src/ColumnWriter.cc]: same here and below

File: c++/src/ByteRLE.cc [hunk 3]
```diff
@@ -55,7 +55,7 @@ namespace orc {
     virtual uint64_t getBufferSize() const override;
 
     /**
-     * Flushing underlying BufferedOutputStream
+     * Flush underlying BufferedOutputStream.
     */
```
[the remark this answers is not in the document]
Comment (fangzheng) [reply c++/src/ByteRLE.cc]: Done.

File: c++/src/ColumnWriter.cc (hunk 1 above)
Comment (fangzheng) [reply c++/src/ColumnWriter.cc]: Done.

File: c++/src/ColumnWriter.cc (hunk 2 above)
Comment (fangzheng) [reply c++/src/ColumnWriter.cc]: Done.

**Comment by fangzheng** [comment]
Thanks, Gang! All the codes are fixed now.

**Comment by majetideepak** [comment]
@fangzheng can you please resolve the conflict? Thanks.

**Pull request merged**